assets/src/scripts/components/hydrograph/index.js

```diff
@@ -49,21 +51,93 @@ const plotDataLine = function (elem, {visible, lines, tsDataKey, xScale, yScale}
         .y(d => yScale(d.value));
 
     for (let line of lines) {
-        elem.append('path')
-            .datum(line.points)
-            .classed('line', true)
-            .classed('approved', line.classes.approved)
-            .classed('estimated', line.classes.estimated)
-            .attr('data-title', tsDataKey)
-            .attr('id', `ts-${tsDataKey}`)
-            .attr('d', tsLine);
+        if (line.classes.dataMask === null) {
+            elem.append('path')
+                .datum(line.points)
+                .classed('line', true)
+                .classed('approved', line.classes.approved)
+                .classed('estimated', line.classes.estimated)
+                .attr('data-title', tsDataKey)
+                .attr('id', `ts-${tsDataKey}`)
+                .attr('d', tsLine);
+        }
+        else {
+            let maskCode = line.classes.dataMask.toLowerCase();
+            let maskDisplayName = MASK_DESC[maskCode].replace(' ', '-').toLowerCase();
+            let xMaskExtent = extent(line.points, d => d.time);
+            let [xDomainStart, xDomainEnd] = xMaskExtent;
+            let [yRangeStart, yRangeEnd] = yScale.domain();
+            let maskGroup = elem
+                .append('g')
+                    .attr('class', `${tsDataKey}-mask-group`);
+
+            maskGroup.append('rect')
+                .attr('x', xScale(xDomainStart))
+                .attr('y', yScale(yRangeEnd))
+                .attr('width', xScale(xDomainEnd) - xScale(xDomainStart))
+                .attr('height', Math.abs(yScale(yRangeEnd)- yScale(yRangeStart)))
+                .attr('class', `mask ${maskDisplayName}-mask`);
+
+            let patternId = tsDataKey === 'compare' ? 'url(#hash-135)' : 'url(#hash-45)';
```

Comment:
I wonder if the hashes should be constants at the module level so that they aren't buried in the code?

Reply:
I think that would be good.